modules/mono/glue/GodotSharp/GodotSharp/Core/NativeInterop/VariantUtils.generic.cs

```diff
@@ -0,0 +1,406 @@
+using System;
+using System.Diagnostics.CodeAnalysis;
+using System.Runtime.CompilerServices;
+
+namespace Godot.NativeInterop;
+
+public partial class VariantUtils
+{
+    private static Exception UnsupportedType<T>() => throw new InvalidOperationException(
+        $"The type is not supported for conversion to/from Variant: '{typeof(T).FullName}'");
```

Comment:
Since this method throws, shouldn't it be `void`? Also may want to add the prefix `Throw`. You can add the `[DoesNotReturn]` attribute but I think the compiler will still complain if the caller does not return, AFAIK it's only used for null flow analysis. 

```suggestion
    [DoesNotReturn]
    private static void ThrowUnsupportedType<T>() => throw new InvalidOperationException(
        $"The type is not supported for conversion to/from Variant: '{typeof(T).FullName}'");
```

Reply:
Oops, this method should not be throwing. The caller throws the returned exception.

Reply:
Oh I misunderstood then. I guess it doesn't really matter either way.

I was going to mention that methods that throw can't be inlined but `CreateFrom` and `ConvertTo` only use it for enums and you are already handling all possible underlying types so it seems to be optimized by the JIT.

I'm not sure about `GenericConversion<T>` though but I guess that's already the slow path.

Reply:
Yes, it's optimized away. It's only included in `CreateFrom` and `ConvertTo` when it will actually throw for sure. Also, even though that scenario should be fixed by the user and should actually result in a compiler error thanks to `[MustBeVariant]`, I rather not have `new InvalidOperationException($"...: '{typeof(T).FullName}'");` be inlined, as it results in more instructions. It should not happen in practice, but to make intentions clear.

Reply:
This PR was merged before I got a chance to change this. It doesn't do any harm, as the exception is thrown any way. Still, a follow-up patch is welcome.

Reply:
Opened https://github.com/godotengine/godot/pull/69191